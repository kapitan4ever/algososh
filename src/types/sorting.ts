import { ElementStates } from "./element-states"

export type TNewArr = {
	item: number
	state: ElementStates
}

export type TInProgress = {
	ascending: boolean
	descending: boolean
}
