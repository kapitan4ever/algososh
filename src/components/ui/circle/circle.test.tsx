import { render } from '@testing-library/react'
import { ElementStates } from '../../../types/element-states'
import { Circle } from './circle'

describe('Circle', () => {
	it('renders with correct text', () => {
		const { getByTestId } = render(<Circle letter='circleExample' />)
		const circle = getByTestId('circle')

		expect(circle).toHaveTextContent('circleExample')
		expect(circle).toMatchSnapshot()
	})
	it('renders without correct text', () => {
		const { getByTestId } = render(<Circle />)
		const circle = getByTestId('circle')

		expect(circle).toBeInTheDocument()
		expect(circle).toMatchSnapshot()
	})
	it('renders with correct text in head', () => {
		const { getByTestId } = render(<Circle head='head' />)
		const circle = getByTestId('circle')

		expect(circle).toHaveTextContent('head')
		expect(circle).toMatchSnapshot()
	})
	it('renders with correct text in tail', () => {
		const { getByTestId } = render(<Circle tail='tail' />)
		const circle = getByTestId('circle')

		expect(circle).toHaveTextContent('tail')
		expect(circle).toMatchSnapshot()
	})
	it('renders with correct text in react element head', () => {
		const { getByTestId } = render(<Circle head={<a>head</a>} />)
		const circle = getByTestId('circle')

		expect(circle).toHaveTextContent('head')
		expect(circle).toMatchSnapshot()
	})
	it('renders with correct text in react element tail', () => {
		const { getByTestId } = render(<Circle tail={<a>tail</a>} />)
		const circle = getByTestId('circle')

		expect(circle).toHaveTextContent('tail')
		expect(circle).toMatchSnapshot()
	})
	it('renders with correct index', () => {
		const { getByTestId } = render(<Circle index={8} />)
		const circle = getByTestId('circle')

		expect(circle).toHaveTextContent('8')
		expect(circle).toMatchSnapshot()
	})
	it('renders with correct isSmall', () => {
		const { getByTestId } = render(<Circle isSmall />)
		const circle = getByTestId('circle-main')

		expect(circle).toHaveClass('small')
		expect(circle).toMatchSnapshot()
	})
	it('renders with correct state Default', () => {
		const { getByText } = render(<Circle state={ElementStates.Default} letter='default' />)
		const circle = getByText('default')

		expect(circle).toHaveTextContent('default')
		expect(circle).toMatchSnapshot()
	})
	it('renders with correct state Modified', () => {
		const { getByText } = render(<Circle state={ElementStates.Modified} letter='modified' />)
		const circle = getByText('modified')

		expect(circle).toHaveTextContent('modified')
		expect(circle).toMatchSnapshot()
	})
	it('renders with correct state Changing', () => {
		const { getByText } = render(<Circle state={ElementStates.Changing} letter='changing' />)
		const circle = getByText('changing')

		expect(circle).toHaveTextContent('changing')
		expect(circle).toMatchSnapshot()
	})
})
