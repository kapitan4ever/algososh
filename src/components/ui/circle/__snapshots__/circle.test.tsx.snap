// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Circle renders with correct index 1`] = `
<div
  class="content "
  data-testid="circle"
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    class="circle   default"
    data-testid="circle-main"
  >
    <p
      class="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
  >
    8
  </p>
  <div
    class="text text_type_input text_color_input mt-4 absolute tail60 element"
  />
</div>
`;

exports[`Circle renders with correct isSmall 1`] = `
<div
  class="circle  small default"
  data-testid="circle-main"
>
  <p
    class="text text_type_circle text_color_input letter"
  />
</div>
`;

exports[`Circle renders with correct text 1`] = `
<div
  class="content "
  data-testid="circle"
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    class="circle   default"
    data-testid="circle-main"
  >
    <p
      class="text text_type_circle text_color_input letter"
    >
      circleExample
    </p>
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle renders with correct text in head 1`] = `
<div
  class="content "
  data-testid="circle"
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head string"
  >
    head
  </div>
  <div
    class="circle   default"
    data-testid="circle-main"
  >
    <p
      class="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle renders with correct text in react element head 1`] = `
<div
  class="content "
  data-testid="circle"
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
  >
    <a>
      head
    </a>
  </div>
  <div
    class="circle   default"
    data-testid="circle-main"
  >
    <p
      class="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle renders with correct text in react element tail 1`] = `
<div
  class="content "
  data-testid="circle"
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    class="circle   default"
    data-testid="circle-main"
  >
    <p
      class="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
  >
    <a>
      tail
    </a>
  </div>
</div>
`;

exports[`Circle renders with correct text in tail 1`] = `
<div
  class="content "
  data-testid="circle"
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    class="circle   default"
    data-testid="circle-main"
  >
    <p
      class="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 string"
  >
    tail
  </div>
</div>
`;

exports[`Circle renders without correct text 1`] = `
<div
  class="content "
  data-testid="circle"
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    class="circle   default"
    data-testid="circle-main"
  >
    <p
      class="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;
